qualtran/qir_example.ipynb

```diff
@@ -0,0 +1,685 @@
+{
+ "cells": [
+  {
+   "cell_type": "code",
+   "execution_count": 42,
+   "outputs": [],
+   "source": [
+    "import numpy as np\n",
+    "import pyqir\n",
+    "import qualtran\n",
+    "from qualtran.bloqs.basic_gates import Hadamard, XGate, YGate, ZGate, Rx, Ry, Rz, TGate, TwoBitSwap, CNOT, Toffoli\n",
+    "from qualtran.bloqs.qft.qft_text_book import QFTTextBook\n",
+    "import pyqir\n",
+    "from pyqir import (\n",
+    "    BasicBlock,\n",
+    "    Builder,\n",
+    "    Context,\n",
+    "    Function,\n",
+    "    Linkage,\n",
+    "    Module,\n",
+    "    ModuleFlagBehavior,\n",
+    ")"
+   ],
+   "metadata": {
+    "collapsed": false
+   }
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 60,
+   "outputs": [],
+   "source": [
+    "import attrs\n",
+    "from typing import Dict\n",
+    "from qualtran import Bloq, BloqBuilder, SoquetT, Signature, Register\n",
+    "\n",
+    "\n",
+    "@attrs.frozen\n",
+    "class Swap(Bloq):\n",
+    "    n: int\n",
+    "\n",
+    "    @property\n",
+    "    def signature(self):\n",
+    "        return Signature.build(x=self.n, y=self.n)\n",
+    "\n",
+    "    def build_composite_bloq(\n",
+    "            self, bb: BloqBuilder, *, x: SoquetT, y: SoquetT\n",
+    "    ) -> Dict[str, SoquetT]:\n",
+    "        xs = bb.split(x)\n",
+    "        ys = bb.split(y)\n",
+    "\n",
+    "        for i in range(self.n):\n",
+    "            xs[i], ys[i] = bb.add(CNOT(), ctrl=xs[i], target=ys[i])\n",
+    "        return {\n",
+    "            'x': bb.join(xs),\n",
+    "            'y': bb.join(ys),\n",
+    "        }\n",
```

Comment:
I think these three are good examples to use as test cases for the conversion code in the new `bloq_to_qir_test.py` file